k8s-api/src/main/java/io/enmasse/k8s/api/EventLogger.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.enmasse.k8s.api;
+
+/**
+ * Interface for handling events
+ */
+public interface EventLogger {
+    /**
+     * Log an event.
+     *
+     * @param reason Created, Deleted etc.
+     * @param message Descriptive message
+     * @param type Severity: Normal, Error, Warning etc.
+     * @param objectKind Kind of object involved in event
+     * @param objectName Name of object involved in event
+     */
+    void log(String reason, String message, String type, String objectKind, String objectName);
```

Comment:
Is it worth creating constants for the even kind "Normal", "Warning" rather than having them subject to typos where this method is called?  The objectKind values should also probably come from constants though not defined here.

Reply:
I'll add for Reason as well